math/src/main/java/com/powsybl/math/timeseries/TimeSeries.java

```diff
@@ -9,29 +9,284 @@
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.JsonToken;
+import com.google.common.base.Stopwatch;
+import com.google.common.primitives.Doubles;
 import com.powsybl.commons.json.JsonUtil;
+import gnu.trove.list.array.TDoubleArrayList;
+import org.slf4j.LoggerFactory;
+import org.threeten.extra.Interval;
 
-import java.io.IOException;
-import java.io.UncheckedIOException;
-import java.io.Writer;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Objects;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
-public interface TimeSeries<P extends AbstractPoint> extends Iterable<P> {
+public interface TimeSeries<P extends AbstractPoint, T extends TimeSeries<P, T>> extends Iterable<P> {
 
     TimeSeriesMetadata getMetadata();
 
     Stream<P> stream();
 
     Iterator<P> iterator();
 
+    List<T> split(int n);
+
+    static <P extends AbstractPoint, T extends TimeSeries<P, T>> List<List<T>> split(List<T> timeSeriesList, int n) {
+        Objects.requireNonNull(timeSeriesList);
+        List<List<T>> splitList = new ArrayList<>(n);
+        for (int i = 0; i < n; i++) {
+            splitList.add(new ArrayList<>(timeSeriesList.size()));
+        }
+        for (T timeSeries : timeSeriesList) {
+            List<T> split = timeSeries.split(n);
+            for (int i = 0; i < n; i++) {
+                splitList.get(i).add(split.get(i));
+            }
+        }
+        return splitList;
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(Path file) {
+        return parseCsv(file, TimeSeriesConstants.DEFAULT_SEPARATOR);
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(String csv, char separator) {
+        try (BufferedReader reader = new BufferedReader(new StringReader(csv))) {
```

Comment:
Reader reader = new BufferedReader(...)

Reply:
I need a BufferedReader here because it contains additional convenient methods like readLines() compare to Reader.